### PR TITLE
dirac: avoid compiling with new GCCs (work in progress)

### PR DESCRIPTION
**Pull request by L2G**
dirac is a GNU-style build that hasn't been updated in so long that it may be a lost cause now. But I've made a branch with my attempts to make it work again.

The pull request is mainly for code-reviewing purposes; it is _NOT_ by any means ready to merge at this writing.

### CODE REVIEW

**Comment by L2G** [comment]
I'm getting good results with apple-gcc42. I just need to work on the formula some more to add some form of `fails_with :clang`.

File: Library/Formula/dirac.rb [hunk 1]
```diff
@@ -1,20 +1,39 @@
...
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
 
   fails_with :llvm do
     build 2334
   end
 
+  patch :DATA
```
Comment (dunn) [inline Library/Formula/dirac.rb]: Could this be an `inreplace` instead?

Comment (L2G) [inline Library/Formula/dirac.rb]: Yes, sorry, I was under the impression that patches were preferred over inreplaces.

Comment (L2G) [inline Library/Formula/dirac.rb]: I see the Formula Cookbook recommends using `inreplace` when it's certain or at least very unlikely that a patch would be accepted upstream.

**Comment by L2G** [comment]
I see that the Jenkins build failed with `gcc` installed. On my Mavericks system, it succeeds but with `apple-gcc42` installed. I need to find a way to add a `fails_with :gcc` that still permits the right version(s) of GCC.

**Comment by jacknagel** [comment]
>build uses compiler flags not known to clang

Which flags? We have the ability to [filter out flags in the compiler wrapper](https://github.com/Homebrew/homebrew/blob/716e4dc503e2912e3c492739e31e2b0317a1dfc0/Library/ENV/4.3/cc#L134) in the build environment.

**Comment by L2G** [comment]
One of them was `-nologo`. I'll have to take out the automake/autoconf stuff to refresh my memory about the other one. (Right now the Clang build fails with automake/autoconf: the configure script dies with "Unsupported Compiler version".)

**Comment by L2G** [comment]
Is there a way to specify known, good versions of GCC in the formula (as opposed to known, bad versions)? Now I'm seeing Jenkins give these tests a false "success" because of this:

```
==> SKIPPING: dirac
dirac cannot be built with any available compilers.
To install this formula, you may need to:
  brew install gcc
```

**Comment by jacknagel** [comment]
What do the gcc compilation errors look like? Maybe we can fix them.

**Comment by mistydemeo** [comment]
I was able to fix the gcc-4.9 compilation errors using this patch from their git repo: http://diracvideo.org/git?p=dirac-research.git;a=patch;h=c99cf8d2cb1e9e69b464f4df2dcae9f9baab49dc

**Comment by mistydemeo** [comment]
There have been two commits in their repo since the 1.0.2 release, that one and this: http://diracvideo.org/git?p=dirac-research.git;a=patch;h=88fae6224b54ecf92ac6e3ae5083c3f352adc798

**Comment by mistydemeo** [comment]
clang isn't working for me due to a bad assumption that a compiler beginning with "cl" is an MS compiler:

```sh
dnl -------------------------------------------------------------------------
dnl Check if compiling using MS VC++ compiler in msys environment on Windows
dnl -------------------------------------------------------------------------
use_msvc=no
AC_SUBST([LIBFLAGS],[])
AC_SUBST([LIBEXT],[])
case "$CXX" in
		cl*|CL*)
		CXXFLAGS="-nologo -W1 -EHsc -DWIN32"
		if test x"$enable_shared" = "xyes"; then
		    LIBEXT=".dll";
		    LIBFLAGS="-DLL -INCREMENTAL:NO"
			CXXFLAGS="$CXXFLAGS -D_WINDLL"
		else
		    LIBEXT=".lib";
		    LIBFLAGS="-lib"
		fi
		RANLIB="echo"
		use_msvc=yes
		;;
	*)
		;;
esac
```

**Comment by mistydemeo** [comment]
Actually, it looks like the bad flags are only passed because it thinks it's using a Windows compiler.

**Comment by mistydemeo** [comment]
Fixed - thank you so much for the report and the helpful diagnosis!

**Comment by L2G** [comment]
So the other changes for `brew style` that were also in there…?

**Comment by mistydemeo** [comment]
Applied as b7ca5ecfa80fad4581ce5c5f1d59714375d95772.

**Comment by L2G** [comment]
Whew.

**Pull request closed without merging**